app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -180,53 +179,18 @@ class LuciBuildService {
     String cipdVersion = 'refs/heads/${pullRequest.base!.ref!}';
     cipdVersion = branches.contains(cipdVersion) ? cipdVersion : config.defaultRecipeBundleRef;
 
+    final List<Future<Request>> futures = <Future<Request>>[];
     for (Target target in targets) {
-      final github.CheckRun checkRun = await githubChecksUtil.createCheckRun(
-        config,
-        target.slug,
-        sha,
-        target.value.name,
-      );
-
-      final github.RepositorySlug slug = pullRequest.base!.repo!.slug();
-
-      final Map<String, dynamic> userData = <String, dynamic>{
-        'builder_name': target.value.name,
-        'check_run_id': checkRun.id,
-        'commit_sha': sha,
-        'commit_branch': pullRequest.base!.ref!.replaceAll('refs/heads/', ''),
-      };
-
-      final Map<String, List<String>> tags = <String, List<String>>{
-        'github_checkrun': <String>[checkRun.id.toString()],
-      };
-
-      final Map<String, Object> properties = target.getProperties();
-      properties.putIfAbsent('git_branch', () => pullRequest.base!.ref!.replaceAll('refs/heads/', ''));
-
-      final List<String>? labels = extractPrefixedLabels(pullRequest.labels, githubBuildLabelPrefix);
-
-      if (labels != null && labels.isNotEmpty) {
-        properties[propertiesGithubBuildLabelName] = labels;
-      }
-
-      requests.add(
-        Request(
-          scheduleBuild: _createPresubmitScheduleBuild(
-            slug: slug,
-            sha: pullRequest.head!.sha!,
-            //Use target.value.name here otherwise tests will die due to null checkRun.name.
-            checkName: target.value.name,
-            pullRequestNumber: pullRequest.number!,
-            cipdVersion: cipdVersion,
-            userData: userData,
-            properties: properties,
-            tags: tags,
-            dimensions: target.getDimensions(),
-          ),
+      futures.add(
+        _scheduleTryBuild(
+          target: target,
+          sha: sha,
+          pullRequest: pullRequest,
+          cipdVersion: cipdVersion,
         ),
       );
     }
+    final List<Request> requests = await Future.wait(futures);
```

Comment:
I doubt this will help since there is no async operation in `_scheduleTryBuild` at all

Reply:
githubChecksUtil.createCheckRun is async, and seems to take about 1 second

Reply:
Yeah, you are right.

One concern is the `second rate limit`. With this change, I expect the API will be called more frequently within a short time and will increase the chance to hit the second rate limit.

IIUC, this PR aims to reduce the show up latency of these check runs, but doesn't seem to help with the LUCI builds scheduling/execution time.

Reply:
For the secondary rate limit, my understanding is that is only an issue on authentication. This is for repeat calls to an existing client. Or is there some other logic happening under the hood now?

For the LUCI scheduling, that is published to the scheduler-reqeusts topic which should take ~10ms. We could parallelize it if there's interest, but I estimate it's <1% of the framework time.

Reply:
Thanks for the link, I'll close this. However, Cocoon isn't implementing this best practice today, so I filed https://github.com/flutter/flutter/issues/142816 for tracking.